src/backend/opencl/cl2hpp.hpp

```diff
@@ -0,0 +1,18 @@
+/*******************************************************
+ * Copyright (c) 2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wignored-qualifiers"
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include <CL/cl2.hpp>
```

Comment:
wasn't this under platform? The name is confusing because it looks like its a standard header that is provided by Khronos. I don't see a huge advantage of doing it this way.

Reply:
This is cleaner and lets us include cl headers in `.cpp` files where OpenCL objects are needed without including `platform.hpp` in all the files just for OpenCL headers.

Additional benefit: It is lot cleaner to keep `cl2.hpp` related pragma's and inclusion only in once place, else they are scattered and bound to diverge.